src/main/java/Control/Scenario/UC_Auth.java

```diff
@@ -0,0 +1,238 @@
+ */
+public class UC_Auth {
+    private DbProvider db;
+
+    public UC_Auth(DbProvider dbProvider) {
+        this.db = dbProvider;
+    }
+
+    public ArrayList<T_Address> retrieveAllAddress() { // TODO remove this method later
+        try {
+            return Address.retrieveAll(db.getConn(), db.getPs(), db.getRs());
+        } catch (SQLException e) {
+            CustomLogs.Error(e.getMessage());
+        }
+        return null;
+    }
+
+    /**
+     * Retrieve user and password hash
+     * @param email User to search for
+     * @return {@link Auth} instance, null if email not in database
+     */
+    private Auth retrieveAuthByEmail(final String email) throws SQLException {
+
+        T_User t_user = Model.Database.Interaction.User.retrieveByEmail(db.getConn(), db.getPs(), db.getRs(), email);
+        if (t_user == null) { // isnt email in db?
+            return null;
+        }
+
+        Model.Web.User user = new Model.Web.User(); // TODO differentiate between 2 User classes
+        user.setBeforetitle(t_user.getA_BeforeTitle());
+        user.setFirstname(t_user.getA_FirstName());
+        user.setMiddlename(t_user.getA_MiddleName());
+        user.setLastname(t_user.getA_LastName());
+        user.setPhone(t_user.getA_Phone());
+        user.setEmail(t_user.getA_Email());
+        user.setResidence(t_user.getA_PermanentResidence());
+        user.setUserID(t_user.getA_pk());
+
+        T_Hash t_hash = Hash.retrieveLatest(db.getConn(), db.getPs(), db.getRs(), t_user.getA_pk());
+
+        Auth auth = new Auth();
+        auth.setUser(user);
+        auth.setPassword(t_hash.getA_Value());
+
+        return auth;
+    }
+
+    /**
+     * Create new user
+     * @param auth User data
+     * @return final Api response body
+     */
+    public final JsonResponse createUser(final Auth auth) {
+        JsonResponse jsonResponse = new JsonResponse();
+
+        try {
+            db.beforeSqlExecution();
+
+            if (retrieveAuthByEmail(auth.getUser().getEmail()) != null) {
+                jsonResponse.setMessage("User with this email already exists.");
+                throw new AuthenticationException("User with this email already exists.");
+            }
+
+            // modify User table START
+            Dictionary dict_user = new Hashtable();
+            dict_user.put(T_User.DBNAME_BEFORETITLE, auth.getUser().getBeforetitle());
+            dict_user.put(T_User.DBNAME_FIRSTNAME, auth.getUser().getFirstname());
+            dict_user.put(T_User.DBNAME_MIDDLENAME, auth.getUser().getMiddlename());
+            dict_user.put(T_User.DBNAME_LASTNAME, auth.getUser().getLastname());
+            dict_user.put(T_User.DBNAME_PHONE, auth.getUser().getPhone());
+            dict_user.put(T_User.DBNAME_EMAIL, auth.getUser().getEmail());
+            dict_user.put(T_User.DBNAME_PERMANENTRESIDENCE, auth.getUser().getResidence());
+            dict_user.put(T_User.DBNAME_BLOCKED, false); // TODO autogenerate by database
```

Comment:
databaza ma automaticke pridanie false, ja to trochu poupravim a poslem commit do DCS-190